crates/common/chain/lean/src/service.rs

```diff
@@ -208,13 +231,16 @@ impl LeanChainService {
         Ok(())
     }
 
-    async fn handle_vote(&mut self, vote_item: VoteItem) -> anyhow::Result<()> {
-        let vote = match vote_item {
-            VoteItem::Signed(vote) => {
-                // TODO: Validate the signature.
-                vote.data
-            }
-            VoteItem::Unsigned(vote) => vote,
+    async fn handle_process_vote(
+        &mut self,
+        signed_vote: SignedVote,
+        is_trusted: bool,
+    ) -> anyhow::Result<()> {
+        let vote = if is_trusted {
+            signed_vote.data
+        } else {
+            // TODO: Validate the signature.
+            signed_vote.data
```

Comment:
```suggestion
        if !is_trusted {
            // TODO: Validate the signature.
        }
        let vote = signed_vote.data;
```

The if else seems redundant now